src/gbm/gblinear_model.cc

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright 2019 by Contributors
+ */
+#include <utility>
+#include <limits>
+
+#include "gblinear_model.h"
+
+namespace xgboost {
+namespace gbm {
+
+void GBLinearModel::Save(Json* p_out) const {
+  using WT = std::remove_reference<decltype(std::declval<decltype(weight)>().back())>::type;
```

Comment:
Optional. What do you think about using WeightType or Weight instead of WT? It seems there is only one usage so we can use longer names which are easier to read.